src/NLog/Layouts/Typed/Layout.cs

```diff
@@ -0,0 +1,315 @@
+            {
+                rawValue = _value;
+                return true;
+            }
+
+            if (_layout == null)
+            {
+                rawValue = null;
+                return true;
+            }
+
+            if (_layout.TryGetRawValue(logEvent, out var raw))
+            {
+                var success = TryConvertRawToValue(raw, out var i);
+                rawValue = i;
+                return success;
+            }
+
+            rawValue = null;
+            return false;
+        }
+
+        /// <summary>
+        /// Render to value
+        /// </summary>
+        /// <returns></returns>
+        T IRenderable<T>.RenderToValue(LogEventInfo logEvent)
+        {
+            return RenderToValueInternal(logEvent, null);
+        }
+
+        /// <summary>
+        /// Render to value
+        /// </summary>
+        /// <param name="logEvent"></param>
+        /// <param name="reusableBuilder">if null, default layout render will be used</param>
+        /// <returns></returns>
+        internal T RenderToValueInternal(LogEventInfo logEvent, [CanBeNull] StringBuilder reusableBuilder)
+        {
+            if (_fixedValue)
+            {
+                return _value;
+            }
+
+            if (_layout == null)
+            {
+                return default(T);
+            }
+
+            if (_layout.TryGetRawValue(logEvent, out var raw))
+            {
+                if (TryConvertRawToValue(raw, out var value))
+                {
+                    return value;
+                }
+
+                InternalLogger.Warn("rawvalue isn't a {0} ", _typeNamed);
+            }
+
+            var text = reusableBuilder != null ? RenderAllocateBuilder(logEvent, reusableBuilder) : _layout.Render(logEvent);
+            if (TryParse(text, out var parsedValue))
+            {
+                return parsedValue;
+            }
+
+            InternalLogger.Warn("Parse {0} to {1} failed", text, _typeNamed);
+            return default(T);
+        }
+
+        private static bool TryConvertRawToValue(object raw, out T value)
+        {
+            if (raw == null)
+            {
+                value = default(T);
+                return true;
+            }
+
+            if (raw is T i)
+            {
+                value = i;
+                return true;
+            }
+
+            if (TryConvertTo(raw, out value))
+            {
+                return true;
+            }
+
+            value = default(T);
+
+            return false;
+        }
+
+        private static bool TryGetFixedValue(Layout layout, out T value)
+        {
+            if (layout != null && layout is SimpleLayout simpleLayout && simpleLayout.IsFixedText)
+            {
+                if (!TryParse(simpleLayout.FixedText, out value))
+                {
+                    InternalLogger.Warn("layout with text '{0}' isn't an {1}", simpleLayout.FixedText, _typeNamed);
+                }
+
+                return true;
```

Comment:
Strange that failing to parse a fixed-value, suddenly changes it into a valid type. Now `TryGetRawValue` will return `true`, but actually the fixed-value is invalid.

Reply:
this isn't good indeed. Will fix this

Reply:
fixed